### DISCUSSION TITLE
Exclude a package matched based on matchSourceUrlPrefixes rule

### DISCUSSION THREAD

**Discussion by knuhol** [renovatebot/renovate#25139] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I am having this config for grouping packages which should be updated together:

```json
  "packageRules": [
    {
      "groupName": "react monorepo",
      "description": "Group packages from react monorepo together",
      "matchSourceUrlPrefixes": [
        "https://github.com/facebook/react"
      ]
    }
  ]
```

However, there is a package `eslint-plugin-react-hooks` which I want to exclude from this group and receive separate Renovate PRs for it.

### What you tried so far.

I tried adding `eslint-plugin-react-hooks` into `excludePackageNames`:

```json
  "packageRules": [
    {
      "groupName": "react monorepo",
      "description": "Group packages from react monorepo together",
      "matchSourceUrlPrefixes": [
        "https://github.com/facebook/react"
      ],
      "excludePackageNames": [
        "eslint-plugin-react-hooks"
      ]
    }
  ]
```

However, Renovate keeps including `eslint-plugin-react-hooks` updates into the same PR as the rest of React packages.

Could it be that when the package is matched based on URL rather than its name, `excludePackageNames` doesn't have any effect? Is there any option how to achieve this? I don't wan to use any other matcher because this one is the most precise and everything else will lead to too many exceptions.

**Comment by rarkins** [comment]
This is probably because Renovate's config:recommended preset already ships with a Facebook monorepo rule, so your rule is kind of doing nothing.

Try this instead:

```
  "packageRules": [
    {
      "description": "Exclude eslint plugin from facebook monorepo group",
      "matchPackageNames": [
        "eslint-plugin-react-hooks"
      ],
      "groupName": null
    }
  ]
```
